tslearn/docs/user_guide/matrix-profile.rst

```diff
@@ -0,0 +1,37 @@
+.. _matrix-profile:
+
+Matrix Profile
+==============
+
+The Matrix Profile, :math:`MP`, is a new time series that can be calculated based on an input time series :math:`T` and a subsequence length :math:`m`. :math:`MP_i` corresponds to the minimal distance from the query subsequence :math:`T_{i\rightarrow i+m}` to :math:`T` [1]_. The region around query subsequence is not included when calculating the minimal distances, as these trivial matches would result in distances of :math:`0`. The Matrix Profile is closely related to the distance calculation used to transform time series into their shapelet-transform space, as illustrated below.
```

Comment:
In our implementation, we do not discard from "the region around query subsequence", but just the identity match (match at time `i` for subsequence extracted from time `i`). Do you have a reference that would suggest to remove other neighboring matches too?

Reply:
Yeah I am probably wrong there! It can return 0 if the segment can be matched in another location than the original one. I'll change that!

Reply:
Regarding what is referred to in the literature as the "exclusion zone" for the " self-match or trivial match", the authors recommend setting this to `excl_zone = int(np.ceil(m / 4))` where `m` is the length of the subsequence window. So, say you have a distance profile, `D`, and the index of the `ith` subsequence is `idx`, then you would apply the exclusion zone as:

```
zone_start = max(0, idx - excl_zone)  # This prevents negative indexing
zone_stop = min(D.shape[-1], idx + excl_zone)   # This prevents indexing beyond the length of `D`
D[..., zone_start : zone_stop + 1] = np.inf  # Notice that we need to add `1` since stop indexing is "exclusive" and we need to ensure that the discard region is symmetrical 
```